src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceFacadeImpl.java

```diff
@@ -143,6 +148,20 @@
         }
     }
 
+    private void validateDuplicateEmailProfiles(List<CaseWorkerDomain> caseWorkerRequest) {
+        List<CaseWorkerDomain> duplicateEmailProfiles = new ArrayList<>();
+        caseWorkerRequest.stream()
+                .map(CaseWorkerProfile.class::cast)
+                .collect(groupingBy(CaseWorkerProfile::getOfficialEmail)).values().stream()
+                .filter(duplicateEmailProfile -> duplicateEmailProfile.size() > 1)
+                .forEach(duplicateEmailProfiles::addAll);
+        if (!duplicateEmailProfiles.isEmpty()) {
```

Comment:
maybe we can use `isNotEmpty(duplicateEmailProfiles)` to avoid `!`?

Reply:
Sure. will do. thanks

Reply:
do we need for each here. Not sure  and we can even remove List intialization List<CaseWorkerDomain> duplicateEmailProfiles = new ArrayList<>();

List<CaseWorkerDomain> duplicateEmailProfiles = caseWorkerRequest.stream()
                .map(CaseWorkerProfile.class::cast)
                .collect(groupingBy(CaseWorkerProfile::getOfficialEmail)).values().stream()
                .filter(duplicateEmailProfile -> duplicateEmailProfile.size() > 1).toList()

Reply:
yes, because `.collect(groupingBy(CaseWorkerProfile::getOfficialEmail)).values().stream()` will return `List<CaseWorkerProfile>`

Reply:
I think below . Please try this

     List<CaseWorkerDomain> duplicateEmailProfiles = caseWorkerRequest.stream()
            .map(CaseWorkerProfile.class::cast)
            .collect(groupingBy(CaseWorkerProfile::getOfficialEmail)).values().stream()
            .filter(duplicateEmailProfile -> duplicateEmailProfile.size() > 1)
            .collect(ArrayList::new, List::addAll, List::addAll);